pkg/appgw/requestroutingrules.go

```diff
@@ -314,6 +311,7 @@ func (c *appGwConfigBuilder) getPathRules(cbCtx *ConfigBuilderContext, listenerI
 		if sslRedirect, _ := annotations.IsSslRedirect(ingress); sslRedirect && listenerAzConfig.Protocol == n.HTTP {
 			targetListener := listenerIdentifier{
 				HostName:     listenerID.HostName,
+				HostNames:    listenerID.HostNames,
```

Comment:
you can instead replace this with (same as above)
```go 
targetListener := listenerID
targetListener.FrontendPort = 443
```

Reply:
Sure